realm/realm-library/src/objectServer/java/io/realm/RealmApp.java

```diff
@@ -448,30 +448,106 @@ private void postSuccess(final T result) {
                 handler.post(new Runnable() {
                     @Override
                     public void run() {
-                        callback.onSuccess(result);
+                        callback.onResult(Result.withResult(result));
```

Comment:
You're not using `Result.success` is the result always non-null? it doesn't seem to be the case since you're returning null in a couple of callbacks ...

Reply:
True. I actually didn't think of it, but in this case `Results.withResult(null)` is actually equal to `Results.success()`. The Result class will end up the correct state either way. Is it worth adding the null check here?

Reply:
I think it does make sense if the implementation was in Kotlin, since `withResult` will not accept nullable values. We can annotate it though in Java with `Nullable`

Reply:
Fair point. I'll change it